src/analysis/complex/upper_half_plane/metric.lean

```diff
@@ -0,0 +1,348 @@
+/-
+Copyright (c) 2022 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import analysis.complex.upper_half_plane.topology
+import analysis.special_functions.arsinh
+import geometry.euclidean.inversion
+
+/-!
+# Metric on the upper half-plane
+
+In this file we define a `metric_space` structure on the `upper_half_plane`. The distance is given
```

Comment:
Could you write a bit more here? E.g. I would expect the word "hyperbolic". Maybe also justify why this metric should be an instance (and not, the subtype metric).

Reply:
@jcommelin Done. Could you please have another look?